src/charm.py

```diff
@@ -792,15 +845,17 @@ def _mongos_layer(self) -> Layer:
     @property
     def mongos_initialised(self) -> bool:
         """Check if mongos is initialised."""
-        return "mongos_initialised" in self.app_peer_data
+        return json.loads(self.app_peer_data.get("mongos_initialised", "false"))
 
     @mongos_initialised.setter
     def mongos_initialised(self, value: bool):
         """Set the mongos_initialised flag."""
-        if value:
-            self.app_peer_data["mongos_initialised"] = str(value)
-        elif "mongos_initialised" in self.app_peer_data:
-            del self.app_peer_data["mongos_initialised"]
+        if isinstance(value, bool):
+            self.app_peer_data["mongos_initialised"] = json.dumps(value)
+        else:
```

Comment:
The second case could never be called (it's a property setter, we're ensured that value always exist.
However, with the former code, this would not raise, because it was only checking for the presence of a field, not its value.
```
self.db_initialised = False
assert self.db_initialised
```
Now, we set a value and use it.
The else case in this setter is just to ensure that we're not giving it an invalid value